Restart the app if environment variables change

quis opened the pull request:
Could be confusing why your changes aren’t having any effect.

Someone should probably test this before merging it 😬

cc @edwardhorsford

gemmaleigh commented:
I tested this locally, when I make changes to the .env file and save them, nodemon doesn't restart the app.

quis commented:
Thanks for the testing.

I’ve changed it so that it actually works in b7bf675:

> Doing this using the existing file-extension-based config doesn’t work because nodemon doesn’t undestand that `.env` is a file with no name, and an extension of `env`. So I’ve removed the file-extension-based stuff entirely and rewritten it to use globbing.
>
> Tested locally that:
> - changing `server.js` or `.env` causes the app to restart
> - changing a file inside `node_modules` _doesn’t_ cause the app to restart

nickcolley commented:
I have pulled Chris' change locally and confirmed that changing a file ending with .js, .json and .env triggers nodemon to restart the application.

nickcolley commented:
I think all we need is an addition to the CHANGELOG under 'features'

quis commented:
@nickcolley done and squashed into aaf531c

joelanman commented:
Quick thought, does this affect very first start up after install? The kit creates empty .env at that point, wonder if that triggers restart and if that's a problem

nickcolley commented:
@joelanman no problems from my testing.

36degrees commented:
Rebased against master.

nickcolley commented:
👍 to merge

36degrees commented:
Thanks, @quis!
